File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.py
class Solution(object):
    def findWinners(self, matches):
        """
        :type matches: List[List[int]]
        :rtype: List[List[int]]
        """
        zeroLoses = []
        oneLose = []
        team_map = {}

        for team in matches:
            team_map[team[0]] = team_map.get(team[0], 0) + 0
            team_map[team[1]] = team_map.get(team[1], 0) + 1

        for i, loses in team_map.items():
            if loses == 0:
                zeroLoses.append(i)
            elif loses == 1:
                oneLose.append(i)

        zeroLoses.sort()
        oneLose.sort()

        return [zeroLoses, oneLose]

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cs
public class Solution {
    public IList<IList<int>> FindWinners(int[][] matches) {
        List<int> zeroLoses = new List<int>();
        List<int> oneLose = new List<int>();
        Dictionary<int, int> map = new Dictionary<int, int>();

        foreach (var team in matches) {
            if (!map.ContainsKey(team[0])) {
                map[team[0]] = 0;
            }
            map[team[0]] += 0;

            if (!map.ContainsKey(team[1])) {
                map[team[1]] = 0;
            }
            map[team[1]] += 1;
        }

        foreach (var entry in map) {
            int i = entry.Key;
            int loses = entry.Value;

            if (loses == 0) {
                zeroLoses.Add(i);
            } else if (loses == 1) {
                oneLose.Add(i);
            }
        }

        zeroLoses.Sort();
        oneLose.Sort();

        return new List<IList<int>> { zeroLoses, oneLose };
    }
}